lib/jobs/obm-control.js

```diff
@@ -64,18 +62,33 @@ function obmControlJobFactory(
             assert.ok(node, "Node should exist to run OBM command");
             assert.ok(node.obmSettings, "Node should have OBM settings to run OBM command");
 
-            var settings = _.find(node.obmSettings, function(setting) {
+            self.settings = _.find(node.obmSettings, function(setting) {
                 return setting ? setting.service === self.obmServiceName : false;
             });
 
-            assert.ok(settings, "Node should have OBM settings for service: " +
-                self.obmServiceName);
+            if (self.obmServiceName) {
+                // throw error if the exact obm service doesn't exist
+                assert.ok(self.settings, "Node should have OBM settings for service: " +
+                    self.obmServiceName);
+            } else {
+                //find the default obm service
+                if(node.obmSettings.length === 0){
+                    assert.fail(true,true, "No OBM service assigned to this node");
+                }else if(node.obmSettings.length === 1) {
+                    self.settings = node.obmSettings[0];
+                    logger.info("Defaulting to obmServiceName "+ self.settings.service +
+                    " for node because only one OBM service present.");
+                }else if(node.obmSettings.length > 1){
```

Comment:
actually I take back my comment about it triggering in both cases... just retested after realizing I had not re-introduced the array with zero elements test back into the mix... and discovered I was able to also hit that else statement... fixed in next push.